dashboard/src/reducers/tableOfContentReducer.js

```diff
@@ -21,23 +21,15 @@ const TableOfContentReducer = (state = initialState, action = {}) => {
   const { type, payload } = action;
   switch (type) {
     case GET_TOC_DATA:
-      return {
-        ...state,
-        stack: [payload],
-        searchSpace: payload.files,
-        tableData: payload.files,
-        contentData: payload,
-        isLoading: false,
-      };
-
     case GET_SUB_DIR_DATA:
       return {
         ...state,
-        stack: [...state.stack, payload],
+        stack: type === GET_TOC_DATA ? [payload] : [...state.stack, payload],
         searchSpace: payload.files,
         tableData: payload.files,
-        currData: payload,
+        currData: type === GET_TOC_DATA ? state.currData : payload,
         isLoading: false,
+        contentData: type === GET_TOC_DATA ? payload : state.contentData,
       };
```

Comment:
Ugh:  I didn't notice that one case updated `currData` while the other updated `contentData`.  😢

Your proposed code has some creative use of ternaries, but, while what you propose is OK, I would vote either for reverting it to what you had previously or for something like this:
```
    case GET_TOC_DATA:
    case GET_SUB_DIR_DATA:
      const ret_val = {
        ...state,
        searchSpace: payload.files,
        tableData: payload.files,
        isLoading: false,
      };

      if (type === GET_TOC_DATA) {
        ret_val.stack = [payload];
        ret_val.contentData = payload;
      } else {
        ret_val.stack = [...state.stack, payload];
        ret_val.currData = payload;
      }
      return ret_val;
```
That is, the two cases are different enough that it's probably not worth collapsing them after all.  (Sorry about that!  I thought that they differed by only one field, not three.)  The extra two ternaries which you added, while clever, are distracting because in half the cases they do nothing; so, it would probably yield better code to group them under a single conditional (either splitting the cases entirely or separating the parts which are different from the parts which are the same).

Pick whichever one you think is best....